tokio/src/macros/instrument.rs

```diff
@@ -0,0 +1,120 @@
+cfg_trace! {
+    macro_rules! instrument_resource {
+        (
+            pin_project,
+            $(#[$meta:meta])*
+            // pin project gets confused when this is a `vis`
+            // and does not infer the projection visibility correctly
+            $visibility:ident struct $struct_name:ident {
+                $(
+                $(#[$field_attrs:ident])*
+                $field_vis:vis $field_name:ident : $field_type:ty
+                ),*$(,)+
+            }
+        ) => {
+            pin_project_lite::pin_project! {
+                $(#[$meta])*
+                $visibility struct $struct_name {
+                    __resource_span: tracing::Span,
```

Comment:
prefixing the name _shouldn't_ be necessary here. the `__` prefix is used in some of the tracing procedural macros, to avoid name collisions with identifiers in the user code. however, since these macros are used only in Tokio itself, I think we can just avoid using the name `resource_span`.

the concern is that the `__` prefix will disable unused-field linting, which is kind of a bummer. it would be nice if rustc complained if we used this macro on a type where we then never actually used the span. not a *huge* deal though.